x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/analysis/analyzer/Verifier.java

```diff
@@ -231,6 +236,8 @@ public Verifier(Metrics metrics) {
 
                 failures.addAll(localFailures);
             });
+
+            checkClientSupportsDataTypes(plan, failures, version);
```

Comment:
This check is useful only when NOT doing SELECT * - that is when the consumer specifies each field individually. The two approaches need to be aligned - either we filter all the way or not at all.

Reply:
In a version-gated answer, the `unsigned_long` fields would appear nowhere in SQL catalog results. So `SELECT *` expansion is consistent with this check. If the user "discovers" the fields "out-of-band" though and explicitly inserts an unsigned_long field in the query, that'll be caught by this check.
Again, I think we do this for other "unsupported" field types too (just not version-gated).

To make sure: I think aligning is great, just not sure on how to improve this, specifically.

Reply:
Similar to the above - worth discussing.
My take is to currently filter things for * but return unsigned long as UNSUPPORTED for old clients.